packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -120,6 +145,237 @@ export default class VTable {
         );
     }
 
+    /**
+     * Apply the given table format to this virtual table
+     * @param format Table format to apply
+     */
+    applyFormatAndStyle(format: TableFormat) {
+        if (!this.table) {
+            return;
+        }
+        format = { ...DEFAULT_FORMAT, ...format };
+        this.table.style.borderCollapse = 'collapse';
+        this.setBordersType(format);
+        this.setColor(format);
+        this.setFirstColumnFormat(format);
+        this.setHeaderRowFormat(format);
+        this.saveTableInfo(this.table, format);
+    }
+
+    /**
+     * Set color to the table
+     * @param format
+     */
+    private setColor(format: TableFormat) {
+        const color = (index: number) => (index % 2 === 0 ? format.bgColorEven : format.bgColorOdd);
+        const { hasBandedRows, hasBandedColumns, bgColorOdd, bgColorEven } = format;
+        const shouldColorWholeTable = !hasBandedRows && bgColorOdd === bgColorEven ? true : false;
+        this.cells.forEach((row, index) => {
+            row.forEach(cell => {
+                if (cell.td) {
+                    if (hasBandedRows) {
+                        const backgroundColor = color(index);
+                        cell.td.style.backgroundColor = backgroundColor;
+                    } else if (shouldColorWholeTable) {
+                        cell.td.style.backgroundColor = format.bgColorOdd;
+                    } else {
+                        cell.td.style.backgroundColor = null;
+                    }
+                }
+            });
+        });
+        if (hasBandedColumns) {
+            this.cells.forEach(row => {
+                row.forEach((cell, index) => {
+                    const backgroundColor = color(index);
+                    if (cell.td && backgroundColor) {
+                        cell.td.style.backgroundColor = backgroundColor;
+                    }
+                });

```

Comment:
My understanding is hasBandedRows and hasBandedColumns are actually conflict, right? When they are both true, bandedColumns will overwrite bandedRows.

In that case, we should not have both of them, but use a enum instead. Because two booleans has 4 combinations, but "true, true" is not valid, we only allow 3 combinations

Reply:
Then these two loops can be merged into one